engine/src/chunk.cpp

```diff
@@ -5673,6 +5673,89 @@ MCTextChunkIterator::MCTextChunkIterator(Chunk_term p_chunk_type, MCStringRef p_
         MCLocaleBreakIteratorRelease(break_iterator);
 }
 
+// AL-2015-02-10: [[ Bug 14532 ]] Add text chunk iterator constructor for restricted range chunk operations.
+MCTextChunkIterator::MCTextChunkIterator(Chunk_term p_chunk_type, MCStringRef p_text, MCRange p_restriction)
+{
+    /* UNCHECKED */ MCStringCopy(p_text, text);
+    type = p_chunk_type;
+    
+    if (type == CT_CHARACTER && (MCStringIsNative(text) || (MCStringIsSimple(text) && MCStringIsUncombined(text))))
+        type = CT_CODEUNIT;
+    
+    MCBreakIteratorRef break_iterator;
+    
+    break_iterator = nil;
+    sp = nil;
+    range = MCRangeMake(p_restriction . offset, 0);
+    length = p_restriction . length == UINDEX_MAX ? MCStringGetLength(text) : p_restriction . offset + p_restriction . length;
+    // AL-2014-10-24: [[ Bug 13783 ]] Set exhausted to true if the string is immediately exhausted
+    exhausted = (p_restriction . length == 0 || p_restriction . offset>= MCStringGetLength(text));
+    first_chunk = true;
+    break_position = 0;
+    delimiter_length = 0;
+    
+    switch (type)
+    {
+        case CT_TOKEN:
+        {
+            MCAutoStringRef t_substring;
+            MCStringCopySubstring(text, p_restriction, &t_substring);
+            MCValueAssign(text, *t_substring);
+            sp = new MCScriptPoint(text);
+        }
+            break;
+        case CT_CHARACTER:
+        case CT_SENTENCE:
+        {
+            MCAutoStringRef t_substring;
+            MCStringCopySubstring(text, p_restriction, &t_substring);
```

Comment:
The problem is that our break iterators need to create an icu::String object which (I believe - but it has been a while) takes ownership of the UTF-16 data passed to it. I'll take another look to see if there is a way to do it and have it just point into the StringRef's buffer directly (but this will require the guarantee that the string won't get native-ised for the lifetime of the iterator...).